File: api/propose.py
from flask import Flask, request, jsonify

# Flaskアプリのインスタンスを作成
app = Flask(__name__)

# `/api/propose` へのリクエストをこの関数で処理する
# Vercelがファイル名（propose.py）をURL（/api/propose）に対応させます


@app.route('/', methods=['GET', 'POST'])
def propose_combination():
    # TODO: 本来はここで「ナップサック問題」のアルゴリズムを動かす

    # クエリパラメータから残額を取得してみる
    # (例: .../api/propose?remaining=350)
    remaining_balance = request.args.get('remaining', default=0, type=int)

    # ダミーの提案を返す
    suggestion = {
        "remaining": remaining_balance,
        "suggestion": [
            {"name": "おにぎり", "price": 140},
            {"name": "お茶", "price": 160}
        ],
        "total": 300
    }

    # JSON形式で結果を返す

    return jsonify(suggestion)

File: api/propose.js
// Vercelはデフォルトで (req, res) を受け取る関数
export default function handler(req, res) {
  // GETリクエスト以外はエラーを返す (もしGET専用にする場合)
  if (req.method !== "GET") {
    res.setHeader("Allow", ["GET"]);
    return res.status(405).end(`Method ${req.method} Not Allowed`);
  }

  // クエリパラメータから残額を取得

  const remaining_balance = parseInt(req.query.remaining || "0", 10);

  const suggestion = {
    remaining: remaining_balance,
    suggestion: [
      { name: "おにぎり", price: 140 },
      { name: "お茶", price: 160 },
    ],
    total: 300,
  };

  // JSON形式で結果を返す
  // ステータス200（OK）でJSONを送信します
  res.status(200).json(suggestion);
}
